README.rst

```diff
@@ -252,12 +252,12 @@ The following options can be configured on the server:
     storage.redis.sentinel.password                                                                                                                                                                                                                                                                                                                                Password for authenticating to Redis Sentinels.
     storage.redis.sentinel.username                                                                                                                                                                                                                                                                                                                                Username for authenticating to Redis Sentinels.
     storage.redis.tls.truststorefile                                                                                                                                                                                                                                                                                                                               PEM file containing the trusted CA certificate(s) for authenticating remote Redis servers. Can only be used when connecting over TLS (use 'rediss://' as scheme in address).
-    storage.sql.connection                                                                                                                                                                                                                                                                                                                                         Connection string for the SQL database. If not set, it defaults to a SQLite database stored inside the configured data directory
+    storage.sql.connection                                                                                                                                                                                                                                                                                                                                         Connection string for the SQL database. If not set, it defaults to a SQLite database stored inside the configured data directory. If specifying a SQLite database, make sure to enable foreign keys with the '_foreign_keys=on' query parameter.
```

Comment:
shared cache is discouraged (https://www.sqlite.org/sharedcache.html), but appearantly the only way to make sqlite3 behave in Go (https://github.com/mattn/go-sqlite3/issues/632), since `database/sql` tries to open multiple connections anyway (and GORM uses the stdlib package). Otherwise, we'll end up with `database busy` errors when load increases (and multiple connections are opened). I think we've seen that error before.

This then touches isolation semantics:
- I used "SELECT FOR UPDATE" to avoid concurrent updates to the discovery service (messing up the timestamping), but this isn't supported by SQLite.
- It does, however, support table-level transactions. There can only be 1 write-TX for each table active.
- A TX, however, does not become a write-TX until the first actual write operation, so the current `updateTimestamp()` that uses `SELECT FOR UDPATE`, `UPDATE` (latter using the result from the first) does not prevent current modification of the timestamps.
- Solution:
  1.  use a `SELECT FOR UPDATE` to make sure there's a row-level lock for databases that support it. Throw away the result. 
  2. rewrite the timestamp-update query to increment in SQL, rather than using the previous `result + 1`
  3. then query the timestamp to return it from the function

Reply:
What about a note: don't use in-memory unless for test/debug. Don't use SQLite as discovery server. Using SQLite in production is discouraged.

Reply:
I'll add a note SQLite is discouraged in production environments.